### ISSUE TITLE
iex freezes after reporting error from incorrectly capitalized module name

### ISSUE THREAD

**Issue by robss2020** [elixir-lang/elixir#13249]
### Elixir and Erlang/OTP versions

IEx 1.15.7 (compiled with Erlang/OTP 26)

### Operating system

Windows 10

### Current behavior

If in an app called CounterApp (you can fetch an example here if you want `git clone https://github.com/robss2020/SimpleCounterGenServer` then cd into the directory and start it with `iex -S mix`)  I mistype it in iex like this:

iex(2)> COunterApp.Counter.value

Then rather than fail and let me continue, iex reports the error but then freezes, not letting me continue the session.

The first example below shows the correct behavior for an incorrectly capitalized function, letting me continue the session.

```
Interactive Elixir (1.15.7) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> CounterApp.counter.value #incorrectly capitalized child process
** (UndefinedFunctionError) function CounterApp.counter/0 is undefined (module CounterApp is not available)
    CounterApp.counter()
    iex:1: (file)
iex(1)> CounterApp.Counter.value #corrected
0
iex(2)> CounterApp.Counter.increment
:ok
iex(3)> CounterApp.Counter.value
1
iex(4)> CounterApp.Counter.reset
:ok
iex(5)> CounterApp.Counter.value
0
iex(6)> CounterApp.Counter.increment
:ok
iex(7)> CounterApp.Counter.value
1
iex(8)> COunterApp.Counter.increment #has a typo

22:50:13.423 [error] beam\beam_load.c(180): Error loading module 'Elixir.COunterApp.Counter':
  module name in object code is 'Elixir.CounterApp.Counter'
```

I was not able to continue after the last error, since iex does not let me continue to enter lines, it freezes and required me to end the session.

![image](https://github.com/elixir-lang/elixir/assets/70531052/9f86ef85-52aa-42dc-9faa-8beea0a127ef)

### Expected behavior

The error message is good but IEX should return with the next prompt so the user can correct the typo, it should not freeze in this scenario.

**Comment by sabiwara** [comment]
I believe this is due to a known upstream [issue](https://github.com/erlang/otp/issues/7725) in OTP26, should be fixed by https://github.com/erlang/otp/pull/7755.

**Comment by josevalim** [comment]
Correct. Upstream and fixed in upcoming 26.3.
